hera_cal/tempcal.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 the HERA Project
+# Licensed under the MIT License
+
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+import os
+import copy
+from six.moves import map, range
+from pyuvdata import UVData, UVCal, utils as uvutils
+
+from .vis_clean import VisClean
+from . import utils
+from .datacontainer import DataContainer
+from . import version, io
+
+
+class TempCal(VisClean):
```

Comment:
I don't see the point of making this a subclass of VisClean... you're not really using any of the build in functions in VisClean.

I think this code would be a lot more modular and reusable if so much stuff wasn't stored under the hood rather than transparently returned. I'd strongly recommend starting with functions at the top level that take DataContainers rather than having to init an object to use these functions. Given the number of inputs and outputs you have right now, it seems like you should just be returning things rather than opaquely storing them in the TempCal object. 

If you need to write standard ways of chaining things together, write a run function.

Reply:
Well you can see that it inits from VisClean, so it inherits all of VisClean partial IO, metadata storing, on-the-fly input calibration, and data writing... which is kinda a lot.

But I agree that its useful to have top-level functions, although I think I'd prefer them to operate on numpy arrays. I'll add something and then make the `TempCal` object wrap around that. The `TempCal` object is a little thin currently b/c there are some features we haven't incorporated that might be important in the future (like calibration given actual temperature measurements)